projects/plugins/super-cache/wp-cache.php

```diff
@@ -3080,35 +3080,85 @@ function wpsc_get_htaccess_info() {
 	$gziprules =  "<IfModule mod_mime.c>\n  <FilesMatch \"\\.html\\.gz\$\">\n    ForceType text/html\n    FileETag None\n  </FilesMatch>\n  AddEncoding gzip .gz\n  AddType text/html .gz\n</IfModule>\n";
 	$gziprules .= "<IfModule mod_deflate.c>\n  SetEnvIfNoCase Request_URI \.gz$ no-gzip\n</IfModule>\n";
 
-	$vary_header = $cache_control_header = '';
+	// Default headers.
+	$headers = array(
+		'Vary'          => 'Accept-Encoding, Cookie',
+		'Cache-Control' => 'max-age=3, must-revalidate',
+	);
 
-	if ( defined( 'WPSC_VARY_HEADER' ) ) {
-		if ( WPSC_VARY_HEADER != '' ) {
-			$vary_header = WPSC_VARY_HEADER;
-		}
-	} else {
-		$vary_header = 'Accept-Encoding, Cookie';
-	}
-	if ( defined( 'WPSC_CACHE_CONTROL_HEADER' ) ) {
-		if ( WPSC_CACHE_CONTROL_HEADER != '' ) {
-			$cache_control_header = WPSC_CACHE_CONTROL_HEADER;
-		}
-	} else {
-		$cache_control_header = 'max-age=3, must-revalidate';
+	// Allow users to override the Vary header with WPSC_VARY_HEADER.
+	if ( defined( 'WPSC_VARY_HEADER' ) && ! empty( WPSC_VARY_HEADER ) ) {
+		$headers['Vary'] = "'" . addcslashes( WPSC_VARY_HEADER, "'" ) . "'";
 	}
-	$headers_text = "";
-	if ( $vary_header != '' ) {
-		$headers_text .= "  Header set Vary '$vary_header'\n";
-	}
-	if ( $cache_control_header != '' ) {
-		$headers_text .= "  Header set Cache-Control '$cache_control_header'\n";
+
+	// Allow users to override Cache-control header with WPSC_CACHE_CONTROL_HEADER
+	if ( defined( 'WPSC_CACHE_CONTROL_HEADER' ) && ! empty( WPSC_CACHE_CONTROL_HEADER ) ) {
+		$headers['Cache-Control'] = WPSC_CACHE_CONTROL_HEADER;
 	}
+
+	// Allow overriding headers with a filter.
+	$headers = apply_filters( 'wpsc_htaccess_mod_headers', $headers );
+
+	// Combine headers into a block of text.
+	$headers_text = join(
+		"\n",
+		array_map(
+			function ( $key, $value ) {
+				return "  Header set $key $value";
```

Comment:
I tested applying the filter `wpsc_htaccess_mod_header`, returning this:

```php
$headers['Cache-Control'] = "max-age=300, private";
$headers['Accept-Ranges'] = "bytes";
return $headers;
```

That prints the following to the generated `.htaccess` file:
```
<IfModule mod_headers.c>
  Header set Vary Accept-Encoding, Cookie
  Header set Cache-Control max-age=300, private
  Header set Accept-Ranges bytes
</IfModule>
```
See how it doesn't contain the single quotes around the directives values? This results in a 500 error for me. It should correctly print this:
```
<IfModule mod_headers.c>
  Header set Vary 'Accept-Encoding, Cookie'
  Header set Cache-Control 'max-age=300, private'
  Header set Accept-Ranges 'bytes'
</IfModule>
```

I think the cleanest approach would be to keep all directives unquoted initially and quote them in the final `array_map` here, like this:
```php
// Combine headers into a block of text.
$headers_text = join(
	"\n",
	array_map(
		function ( $key, $value ) {
			return "  Header set $key '$value'"; // < added single quotes around the $value
		},
		array_keys( $headers ),
		array_values( $headers )
	)
);
```

Reply:
Thanks for the feedback. I'm not sure why it didn't upset Apache on my test systems, but I have added quotes as you suggested in dd73b7e5fe for certainty.